interactive_engine/src/executor/store/src/v2/ffi.rs

```diff
@@ -0,0 +1,633 @@
+                i64::MAX
+            }
+        }
+    }
+}
+
+#[no_mangle]
+pub extern fn GetPropertyAsFloat(property_handle : PropertyHandle, error: *mut ErrorHandle) -> f32 {
+    unsafe {
+        let handler = &*(property_handle as *const PropertyImpl);
+        match handler.get_property_value() {
+            PropertyValue::Float(data) => {
+                *data
+            }
+            PropertyValue::Null => {
+                let error_hdl = Box::new(Error::Internal("None Record!".parse().unwrap()));
+                *error = Box::into_raw(error_hdl) as ErrorHandle;
+                f32::MAX
+            }
+            _ => {
+                let error_hdl = Box::new(Error::Internal("Wrong Record Type Float!".parse().unwrap()));
+                *error = Box::into_raw(error_hdl) as ErrorHandle;
+                f32::MAX
+            }
+        }
+    }
+}
+
+#[no_mangle]
+pub extern fn GetPropertyAsDouble(property_handle : PropertyHandle, error: *mut ErrorHandle) -> f64 {
+    unsafe {
+        let handler = &*(property_handle as *const PropertyImpl);
+        match handler.get_property_value() {
+            PropertyValue::Double(data) => {
+                *data
+            }
+            PropertyValue::Null => {
+                let error_hdl = Box::new(Error::Internal("None Record!".parse().unwrap()));
+                *error = Box::into_raw(error_hdl) as ErrorHandle;
+                f64::MAX
+            }
+            _ => {
+                let error_hdl = Box::new(Error::Internal("Wrong Record Type Double!".parse().unwrap()));
+                *error = Box::into_raw(error_hdl) as ErrorHandle;
+                f64::MAX
+            }
+        }
+    }
+}
+
+#[no_mangle]
+pub extern fn GetPropertyAsString(property_handle : PropertyHandle, error: *mut ErrorHandle) -> StringSlice {
+    unsafe {
+        let handler = &*(property_handle as *const PropertyImpl);
+        match handler.get_property_value() {
+            PropertyValue::String(data) => {
+                StringSlice::new(data.as_ptr(), data.len())
+            }
+            PropertyValue::Null => {
+                let error_hdl = Box::new(Error::Internal("None Record!".parse().unwrap()));
+                *error = Box::into_raw(error_hdl) as ErrorHandle;
+                StringSlice::null()
+            }
+            _ => {
+                let error_hdl = Box::new(Error::Internal("Wrong Record Type Double!".parse().unwrap()));
+                *error = Box::into_raw(error_hdl) as ErrorHandle;
+                StringSlice::null()
+            }
+        }
+    }
+}
+
+/// Error FFIs
+
+#[no_mangle]
+pub extern fn PrintError(error_handle: ErrorHandle) {
+    unsafe {
+        let handler = &*(error_handle as *const Error);
+        handler.what();
+    }
+}
+
+/// Handle Release FFIs
+
+#[no_mangle]
+pub extern fn ReleasePartitionSnapshotHandle(ptr: PartitionSnapshotHandle) {
+    println!("[Rust FFI] <Release PartitionSnapshotHandle>");
```

Comment:
better use same logger module as the project : log4rs

Reply:
temporary for test, remove it later